code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -1113,3 +1133,24 @@
 
 /mob/living/silicon/ai/zMove(dir, feedback = FALSE, feedback_to = src)
 	. = eyeobj.zMove(dir, feedback, feedback_to)
+
+/mob/living/silicon/ai/proc/handle_hearing(datum/source, list/hearing_args)
+	SIGNAL_HANDLER
+	var/atom/user = hearing_args[HEARING_SPEAKER]
+	if(!ai_chat_highlights || user?.name == name)
+		return
+	var/message = hearing_args[HEARING_RAW_MESSAGE]
+	message = ai_chat_highlights.Replace(message, "<span style='color:pink'>$1</span>")
+	hearing_args[HEARING_RAW_MESSAGE] = message
+	var/temp_holder = ai_chat_highlights.name
+	temp_holder = replacetext(temp_holder, "(", "")
+	temp_holder = replacetext(temp_holder, ")", "")
+	var/list/words = splittext(temp_holder, "|")
+	var/found_mention = FALSE
+	for(var/found_word in words)
+		if(findtext(message, found_word))
+			found_mention = TRUE
+			break
```

Comment:
This is a very strange way to run a regex, simply early return if regex.Find(message) returns false.

Reply:
https://www.byond.com/docs/ref/#/regex/proc/Find